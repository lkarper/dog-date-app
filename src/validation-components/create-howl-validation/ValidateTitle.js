import React, { useEffect } from 'react';

const ValidateTitle = (props) => {
    const { title, titleError, setTitleError } = props;

    useEffect(() => {
        if (title.trim().length === 0) {
            setTitleError('You must provide a title. (Max 100 chars.)')
        } else {
            setTitleError('');
        }
    }, [title, setTitleError]);

    if (titleError) {
        return <p id='title-validator'>{titleError}</p>
    }

    return <p id='title-validator'>Title meets requirements. (Max 100 chars.)</p>
}

export default ValidateTitle;