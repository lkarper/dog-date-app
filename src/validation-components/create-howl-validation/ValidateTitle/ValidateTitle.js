import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateTitle = (props) => {
    const { 
        title, 
        titleError, 
        setTitleError, 
    } = props;

    useEffect(() => {
        if (title.trim().length === 0) {
            setTitleError('You must provide a title. (Max 100 chars.)');
        } else {
            setTitleError('');
        }
    }, [title, setTitleError]);

    if (titleError) {
        return (
            <p 
                className='ValidateTitle__validator error'
                id='title-validator'
            >
                {titleError}
            </p>
        );
    }

    return ( 
        <p 
            className='ValidateTitle__validator valid'
            id='title-validator'
        >
            Title meets requirements. (Max 100 chars.)
        </p>
    );
}

ValidateTitle.defaultProps = {
    title: '',
    titleError: 'You must provide a title. (Max 100 chars.)',
    setTitleError: () => {},
};

ValidateTitle.propTypes = {
    title: PropTypes.string.isRequired,
    titleError: PropTypes.string.isRequired,
    setTitleError: PropTypes.func.isRequired,
};

export default ValidateTitle;
