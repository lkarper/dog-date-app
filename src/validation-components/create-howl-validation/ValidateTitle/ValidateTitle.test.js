import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateTitle from './ValidateTitle';

describe('ValidateTitle component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateTitle />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateTitle />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
