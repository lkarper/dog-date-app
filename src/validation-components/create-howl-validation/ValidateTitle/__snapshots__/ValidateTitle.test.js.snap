// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateTitle component renders the UI as expected 1`] = `
<ValidateTitle
  setTitleError={[Function]}
  title=""
  titleError="You must provide a title. (Max 100 chars.)"
>
  <p
    className="ValidateTitle__validator error"
    id="title-validator"
  >
    You must provide a title. (Max 100 chars.)
  </p>
</ValidateTitle>
`;
